OpenRA.Mods.Common/Widgets/TextFieldWidget.cs

```diff
@@ -115,82 +120,180 @@ public override bool HandleKeyPress(KeyInput e)
 			if (!HasKeyboardFocus)
 				return false;
 
-			if ((e.Key == Keycode.RETURN || e.Key == Keycode.KP_ENTER) && OnEnterKey())
-				return true;
+			Func<int> getPrevWhitespaceIndex = () =>
+				Text.Substring(0, CursorPosition).TrimEnd().LastIndexOf(' ') + 1;
+
+			Func<int> getNextWhitespaceIndex = () => {
+				var substr = Text.Substring(CursorPosition);
+				var substrTrimmed = substr.TrimStart();
+				var trimmedSpaces = substr.Length - substrTrimmed.Length;
+				var nextWhitespace = substrTrimmed.IndexOf(' ');
+				if (nextWhitespace == -1)
+					return Text.Length;
+				else
```

Comment:
The if removes, so the body of the else doesn't need to be inside the else (just run it unconditionally).